config/v1/types_infrastructure.go

```diff
@@ -505,9 +505,151 @@ type BareMetalPlatformStatus struct {
 	NodeDNSIP string `json:"nodeDNSIP,omitempty"`
 }
 
+type OpenStackAPIBGPPeer struct {
+	// asn is the Autonomous System number of the peer.
+	//
+	// +kubebuilder:validation:Required
+	ASN string `json:"asn"`
+
+	// ip is the IP address of the peer. It may be either IPv4 or IPv6.
+	//
+	// +kubebuilder:validation:Format=ip
+	// +kubebuilder:validation:Required
+	IP string `json:"ip"`
+}
+
+// OpenStackAPIBGPSpeaker species
+type OpenStackAPIBGPSpeaker struct {
+	// failureDomain is the name of a failure domain which this BGP
+	// configuration applies to. A failure domain with that name must be
+	// defined in the OpenStack platform spec.
+	//
+	// +kubebuilder:validation:Required
+	FailureDomain string `json:"failureDomain"`
+
+	// asn specifies the Autonomous System number to be used by the BGP
+	// speaker.
+	//
+	// +kubebuilder:validation:Required
+	ASN string `json:"asn"`
+
+	// peers is a list of all BGP peers of this speaker. The list must
+	// contain at least one item.
+	//
+	// +kubebuilder:validation:MinItems:=1
+	Peers []OpenStackAPIBGPPeer `json:"peers"`
+}
+
+type OpenStackAPIBGPConfiguration struct {
+	// apiVIPs is a list IP addresses which will be used to reach the API
+	// server. The list must contain at least one address. It may contain
+	// both IPv4 and IPv6 addresses.
+	//
+	// +kubebuilder:validation:Format=ip
```

Comment:
I don't think this works, the vSphere project have this working using a patch, https://github.com/openshift/api/pull/1199/files#diff-8e2a77037de011306a487acc4a7bd2592d519f3415ad6c8b93d191d8cb3e3c32